libs/portals/admin/ids-admin/src/components/Applications/Applications.tsx

```diff
@@ -1,91 +1,233 @@
+  FilterInput,
   GridContainer,
   GridRow,
   Stack,
   Tag,
   Text,
 } from '@island.is/island-ui/core'
-import { Link, useParams } from 'react-router-dom'
-import MockApplications from '../../lib/MockApplications'
-import * as styles from '../TenantsList/TenantsList.css'
+import { Link, useLoaderData, useParams } from 'react-router-dom'
+import React, { useEffect, useState } from 'react'
 import { useLocale } from '@island.is/localization'
 import { m } from '../../lib/messages'
 import { useTenant } from '../../screens/Tenant'
-import { useEffect } from 'react'
 import { replaceParams } from '@island.is/react-spa/shared'
 import { IDSAdminPaths } from '../../lib/paths'
+import * as styles from './Applications.css'
+import { AuthApplicationsList } from './Applications.loader'
 
 const Applications = () => {
+  const originalApplications = useLoaderData() as AuthApplicationsList
   const { tenant } = useParams()
   const { formatMessage } = useLocale()
   const { setNavTitle } = useTenant()
 
+  const [applications, setApplications] = useState<AuthApplicationsList>(
+    originalApplications,
+  )
+  const [inputSearchValue, setInputSearchValue] = useState<string>('')
+
   useEffect(() => {
-    // TODO: Get application by id from backend
     setNavTitle(tenant ? tenant : formatMessage(m.tenants))
   })
 
-  return (
-    <GridContainer>
-      <GridRow rowGap={3}>
-        <Box display={'flex'} justifyContent={'spaceBetween'} columnGap={4}>
-          <Box>
+  const handleSearch = (value: string) => {
+    setInputSearchValue(value)
+
+    if (value.length > 0) {
+      const filteredList = applications.filter((application) => {
+        return (
+          application?.defaultEnvironment.displayName[0].value
+            .toLowerCase()
+            .includes(value.toLowerCase()) ||
+          application.applicationId.toLowerCase().includes(value.toLowerCase())
+        )
+      })
+      setApplications(filteredList)
+    } else {
+      setApplications(originalApplications)
+    }
+  }
+
+  const getHeader = ({ withCreateButton = true }) => {
+    return (
+      <GridRow rowGap={3} marginBottom={'containerGutter'}>
+        <Box
+          width={'full'}
+          display={'flex'}
+          justifyContent={'spaceBetween'}
+          columnGap={'gutter'}
+          alignItems={'center'}
+        >
+          <Stack space={2}>
             <Text variant={'h2'}>{formatMessage(m.applications)}</Text>
-          </Box>
-          <Box>
+            <Text variant={'default'}>
+              {formatMessage(m.applicationsDescription)}
+            </Text>
+          </Stack>
+          {withCreateButton && (
+            <Box>
+              <Button size={'small'}>Create Application</Button>
+            </Box>
+          )}
+        </Box>
+      </GridRow>
+    )
+  }
+
+  return applications.length === 0 ? (
+    <GridContainer>
+      {getHeader({ withCreateButton: false })}
+      <GridRow>
+        <Box
+          display={'flex'}
```

Comment:
~I think it is cleaner to just do `display="flex"` for string values instead of `display={'flex'}` on attributes.
The curly braces in my mind should be used for none string values. It also makes it more easy to spot.